### ISSUE TITLE
Type lost in 1.20.0?

### ISSUE THREAD

**Issue by ModyQyW** [DamianOsipiuk/vue-query#154]
[repo](https://github.com/ModyQyW/vue-query-type-lost) here.

```sh
pnpm install vue-query@1.20.0
```

Restart vscode and open `main.ts`.

![image](https://user-images.githubusercontent.com/31850793/160985351-bcc2f827-5931-40c2-a2cd-71264adba18c.png)

![image](https://user-images.githubusercontent.com/31850793/160985374-e0493f61-9f4f-4e9a-a7d8-fccea0d7428d.png)

![image](https://user-images.githubusercontent.com/31850793/160985385-42385e61-72cc-4bc1-96c4-070291436c94.png)

Everything is fine in 1.19.3.

```sh
pnpm install vue-query@1.19.3
```

Restart vscode and open `main.ts`.

![image](https://user-images.githubusercontent.com/31850793/160985518-dd3508c4-faaf-4626-9353-fcf75f1c7ef2.png)

![image](https://user-images.githubusercontent.com/31850793/160985527-607f69f5-a08a-4c61-abd8-9ce509a487b2.png)

![image](https://user-images.githubusercontent.com/31850793/160985544-4afccf24-2b9b-4430-9e3e-fa30efc3a393.png)

Correct me if I am wrong. Thanks. :D

**Comment by DamianOsipiuk** [comment]
So i have found out that the issue is with the `MaybeRefDeep` type which does not provide proper type narrowing for functions.
I will try to fix it, otherwise `queryClient` constructor will not be able to handle ref unwrapping.

**Comment by DamianOsipiuk** [comment]
@ModyQyW This should be fixed in [1.20.1](https://github.com/DamianOsipiuk/vue-query/releases/tag/v1.20.1)

**Comment by ModyQyW** [comment]
Thanks a lot! :D
